internal/store/serverservice.go

```diff
@@ -179,6 +180,32 @@ func (s *Serverservice) Create(ctx context.Context, serverID uuid.UUID, createCo
 	return err
 }
 
+// AddServer creates a server record in FleetDB
+// @id: required
+// @ip: required
+// @user: required
+// @pwd: required
+func (s *Serverservice) AddServer(ctx context.Context, serverID uuid.UUID, bmcAddr, bmcUser, bmcPass string) error {
+	// Add server
+	server := sservice.Server{UUID: serverID, Name: serverID.String()}
```

Comment:
does the ServerService API have a way to pack this all into a single transaction? Catching an error mid-way and aborting all these individual updates is going to be tedious and erorr-prone.

Reply:
OK, let's do it the hard way to start then. But we will need a method for ourselves that removes a broken server record and gets things cleaned up.

Reply:
Sounds good to me. I will create a ticket and add `DeleteServer` as a follow up PR.

There might be some details I need to think of, eg, if we fails deleting a server which is created successfully, we may want to retry deleting until it is deleted?

Anyway, it may worth a separate PR for cleaning up?